src/Parallelotope.jl

```diff
@@ -0,0 +1,121 @@
+export Parallelotope
+
+"""
+    Parallelotope{N<:Real, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}}
+    <: AbstractZonotope{N}
+
+Type that represents a parallelotope.
+
+### Fields
+
+- `center`     -- vector; center of the parallelotope
+- `directions` -- matrix ``D``; each row is the direction of two corresponding
+                  constraints
+- `offset`     -- vector; each element is the offset of the corresponding constraint
+
+### Notes
+``D_{i} = -D_{i+n}`` for ``i = 1, ..., n`` with ``n`` rows in ``D``
+
+"""
+struct Parallelotope{N<:Real, VN<:AbstractVector{N}, MN<:AbstractMatrix{N}} <: AbstractZonotope{N}
+    center::VN
+    offset::VN
+    directions::MN
+
+    # default constructor with dimension check
+    function Parallelotope(c::VN, D::MN, o::VN) where {N<:Real,
+             VN<:AbstractVector{N}, MN<:AbstractMatrix{N}}
+        @assert length(c) == checksquare(D)
+        @assert length(c) == length(o)
+        return new(c, D, o)
```

Comment:
```suggestion
        return new{VN, MN, VM}(c, D, o)
```
`new` still needs the type parameters.

Reply:
Oops, they were completely wrong, sorry :blush: